### PR TITLE
fix: respect channel_axis in pad_to_size for channels-first images

### PR DESCRIPTION
**Pull request by Nas01010101**
## Summary

`pad_to_size()` ignores `channel_axis` when constructing the pad widths, so it pads the wrong axes for channels-first (CHW) images.

The padding amounts (`top`, `bottom`, `left`, `right`) are computed correctly via `_get_dimension_values`, which is channel-axis-aware. But the `pad_width` tuple passed to `jnp.pad` is hardcoded to channels-last order:

```python
pad_width = ((top, bottom), (left, right), (0, 0))
```

For a CHW image (`channel_axis=0`) this applies the height/width padding to the channel and height axes and leaves width unpadded. Example — padding a `(3, 4, 4)` CHW image to a `(6, 6)` target yields `(5, 6, 4)` instead of the expected `(3, 6, 6)`.

Every other spatial op in this module already branches on `_channels_last(image, channel_axis)`; `pad_to_size` simply missed it.

## Fix

Select the pad-width tuple based on layout:

```python
if _channels_last(image, channel_axis):
    pad_width = ((top, bottom), (left, right), (0, 0))
else:
    pad_width = ((0, 0), (top, bottom), (left, right))
```

The batch-dimension prepend and all other logic are layout-agnostic and unchanged. Channels-last behaviour is identical to before.

## Tests

Adds `dm_pix/_src/pad_to_size_test.py` covering HWC (unchanged), CHW shape, CHW values, and batched CHW. The CHW tests fail before the fix and pass after.

```
4 passed
```
Broader regression (`interpolation_test.py` + the new test): 100 passed, 24 skipped (skips are pre-existing TF-gated tests).

### CODE REVIEW

File: dm_pix/_src/pad_to_size_test.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# limitations under the License.
+"""Tests for pad_to_size with channels-first (CHW) layout."""
+
+from absl.testing import absltest
+from dm_pix._src import augment
+import jax.numpy as jnp
+import numpy as np
+
+
+class PadToSizeChannelsFirstTest(absltest.TestCase):
```
Comment (claudiofantacci) [inline dm_pix/_src/pad_to_size_test.py]: Please move this test under `dm_pix/_src/augment_test.py`.

File: dm_pix/_src/pad_to_size_test.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    """Channels-last (HWC) baseline: output shape must be (H', W', C)."""
+    image = jnp.zeros((4, 4, 3))  # HWC
+    result = augment.pad_to_size(image, target_height=6, target_width=6,
+                                 channel_axis=-1)
+    self.assertEqual(result.shape, (6, 6, 3))
+
+  def test_pad_to_size_chw_shape(self):
+    """Channels-first (CHW) with channel_axis=0: output shape must be (C, H', W').
+
+    Bug: pad_to_size hardcodes pad_width as ((top,bot), (left,right), (0,0)),
```
Comment (claudiofantacci) [inline dm_pix/_src/pad_to_size_test.py]: Remove this comment please.

**Review by claudiofantacci — CHANGES_REQUESTED** [review]
Thanks a lot! 🚀

**Review by claudiofantacci — COMMENTED** [review]
Thanks! It will take a little while to go through and I actually need to drop support from some old Python version.

I'll approve once this is done and then go from there.

In the meantime, thanks again!

**Comment by claudiofantacci** [comment]
Waiting #113 to go through.

**Comment by Nas01010101** [comment]
Sounds good, thanks!!

**Comment by claudiofantacci** [comment]
Can you please rebase on top of main after #113 and re-upload?

**Comment by Nas01010101** [comment]
Of course !

**Review by claudiofantacci — APPROVED** [review]
